test/staged.jl

```diff
@@ -156,7 +156,7 @@ end
 @generated function _g_f_with_inner(x)
     :(y->y)
 end
-@test (_g_f_with_inner(1))(8) == 8
+@test_throws ErrorException _g_f_with_inner(1)
 
```

Comment:
Yes, it's not reasonable to consider a generated function with a closure to have side effects --- the side effects there are in our implementation, not in the user's code.

For backporting, it doesn't matter how "new restriction" is defined. All that matters is whether code can break.